### PR TITLE
Add Ip2cProvider

### PR DESCRIPTION
**Pull request by ganeko**

### CODE REVIEW

**Comment by Nyholm** [comment]
Could you rebase this PR on master?

File: src/Geocoder/Provider/Ip2cProvider.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+namespace Geocoder\Provider;
+
+use Geocoder\Exception\NoResultException;
+use Geocoder\Exception\UnsupportedException;
+use Geocoder\HttpAdapter\HttpAdapterInterface;
+
+/**
+ * @author Ganeko Guereta <ganeko.guereta@gmail.com>
+ */
+class Ip2cProvider extends AbstractProvider implements ProviderInterface
```
Comment (Nyholm) [inline src/Geocoder/Provider/Ip2cProvider.php]: ProviderInterface does not exists. It is called Provider

**Comment by Nyholm** [comment]
You need to modify travis.yml to make sure we run tests for this provider. 
I can also see that there is some stuff missing for being compatible with v3 of the geocoder?

> Any help needed to update this PR to v3?.

Are you asking for help or providing help?

If asking: Run composer update and you will see where PHP complains. There is five or six places you should update. 

If providing: We will be happy to. There is currently a few issues that has some work. See #41 and #40. You may continue my work on #40 if you like.

**Comment by ganeko** [comment]
Hi,

I update my branch and all tests passed. 

Sorry, I was asking about my PR for needed changes but I will check https://github.com/geocoder-php/geocoder-extra/pull/40

**Comment by Nyholm** [comment]
Thank you. I think this PR looks good. 

The only thing missing is updating the table in Readme so we tell the world about the change =)

After that I will be happy to merge.

**Comment by ganeko** [comment]
Oops, sorry about that. Added and committed. Thanks!

**Comment by Nyholm** [comment]
Awesome!

**Pull request merged**